File: 001. La descente/la_descente.py
import sys
import math

# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.


# game loop
while True:
    indice_plus_haute_montagne = 0
    hauteur_plus_haute_montagne = -1
    for i in range(8):
        mountain_h = int(input())  # represents the height of one mountain.
        
        if i == 0 or hauteur_plus_haute_montagne < mountain_h:
            indice_plus_haute_montagne = i
            hauteur_plus_haute_montagne = mountain_h



    # Write an action using print
    # To debug: print("Debug messages...", file=sys.stderr)

    # The index of the mountain to fire on.
    print(indice_plus_haute_montagne)

File: 001. La descente/la_descente.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/

int main()
{

    // game loop
    while (1) {
        int highestMountain = -1;
        int heightHeighestMountain = -1;

        for (int i = 0; i < 8; i++) {
            int mountainH; // represents the height of one mountain.
            cin >> mountainH; cin.ignore();
            cerr << mountainH << endl;
            if (mountainH > heightHeighestMountain) {
                heightHeighestMountain = mountainH;
                highestMountain = i;
            }
        }

        // Write an action using cout. DON'T FORGET THE "<< endl"
        // To debug: cerr << "Debug messages..." << endl;

        cout << highestMountain << endl; // The index of the mountain to fire on.
    }
}